src/Microsoft.AspNetCore.Mvc.RazorPages/DependencyInjection/MvcRazorPagesMvcCoreBuilderExtensions.cs

```diff
@@ -43,24 +38,38 @@ public static IMvcCoreBuilder AddRazorPages(this IMvcCoreBuilder builder)
             }
 
             builder.AddRazorViewEngine();
+
+            AddFeatureProviders(builder);
             AddServices(builder.Services);
 
             builder.Services.Configure(setupAction);
 
             return builder;
         }
 
+        private static void AddFeatureProviders(IMvcCoreBuilder builder)
+        {
+            if (!builder.PartManager.FeatureProviders.OfType<CompiledPageInfoFeatureProvider>().Any())
```

Comment:
I followed the pattern we have in Razor - https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Razor/DependencyInjection/MvcRazorMvcCoreBuilderExtensions.cs#L64. I think this needs to be specific since a user could add a `FeatureProvider<Blah>`, but we still want the built in one to be registered. We're only trying to avoid double registration of the same thing.